### ISSUE TITLE
Key cannot contain a colon

### ISSUE THREAD

**Issue by oerik** [vmihailenco/tagparser#2]
In a particular use case I would like to use a colon inside the key, for example when create a tag for msgpack:
```go
struct Msg {
        ProjectID   uint            `json:"project:id,omitempty" msgpack:"project:id,omitempty"` 
        UserID      uint            `json:"user:id,omitempty" msgpack:"user:id,omitempty"`
}
```

However, the colon is seen as a seperator.  A possible solution to allow this to allow escaping via a \.  For example:
```go
struct Msg {
        ProjectID   uint            `json:"project:id,omitempty" msgpack:"project\\:id,omitempty"` 
        UserID      uint            `json:"user:id,omitempty" msgpack:"user\\:id,omitempty"`
}
```

I have create a PR to implement this, including a testcase.

**Comment by oerik** [comment]
PR https://github.com/vmihailenco/tagparser/pull/3

**Comment by vmihailenco** [comment]
Thanks for reporting and for the PR.

This shoud be fixed by #4 and `msgpack:"'user:id',omitempty"` should be woking.
